AdvancedChangedFiles/AccountProfile.jsx

```diff
@@ -0,0 +1,346 @@
+    };
+
+    init() {      
+        var loaderData = Object.assign({}, this.state.loaderData);
+        loaderData.allowedUsers = loaderData.allowedUsers.concat("Talent");
+        loaderData.isLoading = false;
+        this.setState({ loaderData: loaderData }, function () {
+        }.bind(this));
+    }
+
+    componentDidMount() {
+        this.init();
+        this.loadData();
+    }
+
+    loadData() {
+        var cookies = Cookies.get('talentAuthToken');
+        $.ajax({
+            url: 'https://advancedtalentprofile-e2gkffcccyg0hhdm.australiaeast-01.azurewebsites.net/profile/profile/getTalentProfile',
+            headers: {
+                'Authorization': 'Bearer ' + cookies,
+                'Content-Type': 'application/json'
+            },
+            type: "GET",
+            success: function (res) {           
+                if (res && res.data) {
+                    this.updateWithoutSave(res.data);
+                } else {
+                    console.warn('No data returned from the API.');
+                }
+            }.bind(this),
+            error: function (xhr, status, error) {
+                console.error('Error fetching profile data:', error);
+            }
+        });
+    }
+
+    //updates component's state without saving data
+    updateWithoutSave(newValues) {
+
+        let newProfile = Object.assign({}, this.state.profileData, newValues);
+        this.setState({ profileData: newProfile }, () => {
+
+            // Validate the update
+            if (this.state.profileData.id === newProfile.id) {
+                console.log('Profile updated successfully.');
+            } else {
+                console.warn('Profile update failed.');
+            }
+        });
+    }
+
+    //updates component's state and saves data
+    updateAndSaveData(newValues) {
+        let newProfile = Object.assign({}, this.state.profileData, newValues)
+        this.setState({
+            profileData: newProfile
+        }, this.saveProfile)
+    }
+
+    updateForComponentId(componentId, newValues) {
+        this.updateAndSaveData(newValues)
+    }
+
+    saveProfile() {
+        var cookies = Cookies.get('talentAuthToken');
+        $.ajax({
+            url: 'https://advancedtalentprofile-e2gkffcccyg0hhdm.australiaeast-01.azurewebsites.net/profile/profile/updateTalentProfile',
+            headers: {
+                'Authorization': 'Bearer ' + cookies,
+                'Content-Type': 'application/json'
+            },
+            type: "POST",
+            data: JSON.stringify(this.state.profileData),
+            success: function (res) {
+                console.log(res)
+                if (res.success == true) {
+                    TalentUtil.notification.show("Profile updated sucessfully", "success", null, null)
+                } else {
+                    TalentUtil.notification.show("Profile did not update successfully", "error", null, null)
+                }
```

Comment:
If you are handling boolean in condition,  you can use something like **if (res.success)** only.